halo2_proofs/src/plonk/circuit.rs

```diff
@@ -376,6 +441,10 @@ pub trait Assignment<F: Field> {
         to: Option<Assigned<F>>,
     ) -> Result<(), Error>;
 
+    /// Get challenge value. If current phase is still earlier than requested one, a `None`
+    /// will be returned.
```

Comment:
```suggestion
    /// Queries the value of the given challenge.
    ///
    /// Returns `None` if the current synthesis phase is before the challenge can be queried.
```

Reply:
Updated in [`315d0f0`](https://github.com/zcash/halo2/pull/593/commits/315d0f0e7842d6082d7fdd5a66de30c096dd2b49).